x-pack/plugins/alerting/server/rules_client/lib/should_skip_attributes_update.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { BulkEditSkipReason } from '@kbn/security-solution-plugin/common/detection_engine/rule_management/api/rules/bulk_actions/response_schema';
+import { BulkEditOperation } from '../..';
+import { RawRule } from '../../types';
+
+export const shouldSkipAttributesUpdate = (
+  editOperation: BulkEditOperation,
+  attributes: RawRule,
+  attributesUpdateSkipReasons: BulkEditSkipReason[]
+) => {
+  const { field, operation, value } = editOperation;
+  switch (field) {
+    case 'tags': {
+      if (operation === 'add') {
+        if (value.every((tag) => attributes.tags.includes(tag))) {
```

Comment:
Decided for nesting these if-statements (instead of joining them with AND) to make clear that there could be more cases for the attributes update to be skipped within the `add` operation for tags. Same for `delete`.